### PR TITLE
added solution - 26(2)

### PR DESCRIPTION
**Pull request by glones**

### CODE REVIEW

**Review by vladislavshvets — APPROVED** [review]
Good job!

File: src/main/java/com/internet/shop/controllers/OrderDetailsController.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public class OrderDetailsController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Product> products = orderService.get(USER_ID).getProducts();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/OrderDetailsController.java:22]: you have getUserOrders() method which take userId and returns all orders of this user, but here you are passing userid to into get() method. Let's pass orderId as parameter to this controller and get our order from DB by it

**Review by dmytro-hryhoruk — COMMENTED** [review]
I can't see page for orders of current user

File: src/main/java/com/internet/shop/controllers/CompleteOrderController.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+    private final OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/orders/userOrder.jsp");
+    }
```
[the remark this answers is not in the document]
Comment (glones) [reply src/main/java/com/internet/shop/controllers/CompleteOrderController.java]: removed

File: src/main/java/com/internet/shop/controllers/OrderDetailsController.java (hunk 1 above)
Comment (glones) [reply src/main/java/com/internet/shop/controllers/OrderDetailsController.java:22]: I use method getUserOrders() in my UserOrderController to show an ID of order to client, however in those part I want to show details, so i get an product form my order to show name and price in orderDetails.jsp

File: src/main/java/com/internet/shop/controllers/UserOrderController.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import com.internet.shop.models.Order;
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class UserOrderController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/UserOrderController.java:13]:
```suggestion
public class GetUserOrdersController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/OrderDetailsController.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/com/internet/shop/controllers/OrderDetailsController.java:22]: here let's pass orderId as attribute(from jsp where you are calling this controller) and pass this id to orderService.get() which finds the order by orderId(not by userId)

File: src/main/java/com/internet/shop/controllers/UserOrderController.java (hunk 3 above)
Comment (glones) [reply src/main/java/com/internet/shop/controllers/UserOrderController.java]: corrected

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**